### DISCUSSION TITLE
Introduce a new CommunityToolkit.Maui.Diagnostics package

### DISCUSSION THREAD

**Discussion by bijington** [CommunityToolkit/Maui#468] [New Feature Discussions]
I would like to propose that we introduce a new NuGet package that provides diagnostic support around building applications. We could name it `CommunityToolkit.Maui.Diagnostics`.

Thoughts on its contents are:

## Debug rainbows 

(https://github.com/sthewissen/Xamarin.Forms.DebugRainbows) 
Steven has given his blessing on Twitter and also suggested that David Ortinau might know how to achieve it in a few lines. There is also mention of the alignment grid features (I need to look this up)

## FPS Counter

I recall there was a package somewhere that offered it. I could reach out to the maintainer and see what their thoughts are.

## Additional

I would like us to also investigate how we could easily turn these features off in developers release builds so they can make sure they don't accidentally ship some of these features turned on.

## anything else?

**Comment by VladislavAntonyuk** [comment]
Maybe some logging/analytics/crashes in a more informative/structured format

**Reply by bijington** [reply]
I wouldn't want to suggest we cover analytics but we could certainly see if there are ways to make it easier to collect/structure information

**Comment by pictos** [comment]
@bijington you can find the David's snippet to archive the debug rainbows effect [here](https://github.com/dotnet/maui/wiki/Customizing-Controls-with-Handlers#example-debug-rainbows)

**Reply by bijington** [reply]
Oh great thanks. That really is just a few lines of code!

**Comment by pictos** [comment]
I'll play the _Devil's advocate_ here. 
So I'm not sure if we need to create a new package to support this, as you can see in my [comment](https://github.com/CommunityToolkit/Maui/discussions/468#discussioncomment-3027196) is very easy to create your own debug rainbows. VS also adds more tooling to help devs to debug UI and make sure everything is in its place.

I would say that those features should be implemented in the tooling, maybe the best will be to ask the VS team to put them on the roadmap. And maybe having a package that will be obsolete when VS implements the missing features.

**Reply by bijington** [reply]
I would love for more stuff to go into the tooling for sure! I am also happy with supplying something in the meantime while we wait for the tooling to improve if we feel it provides enough value.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
